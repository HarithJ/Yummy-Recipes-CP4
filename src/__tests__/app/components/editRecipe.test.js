import React from 'react';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson, { shallowToJson } from 'enzyme-to-json';

import EditRecipe from '../../../components/recipes/editRecipe.js'

describe("edit recipe component", () => {
  it("renders properly", () => {
    const editRecipe = shallow(<EditRecipe recipeIngs={[]}/>)
    expect(shallowToJson(editRecipe)).toMatchSnapshot();
  })

  it("has a functions for handling edit", () => {
    const editRecipe = shallow(<EditRecipe recipeIngs={[]}/>)
    const preventDefault = jest.fn();

    expect(editRecipe.instance().handleEditRecipe({ preventDefault }));
    expect(editRecipe.instance().handleAddIng());
    expect(editRecipe.instance().clearInputs());
  })
})
