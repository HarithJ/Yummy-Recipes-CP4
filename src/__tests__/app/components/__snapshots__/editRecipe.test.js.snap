// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit recipe component renders properly 1`] = `
<div>
  <div
    className="col-auto"
  >
    <button
      className="btn btn-outline-warning rounded-circle"
      data-target="#edit-recipe-modal-undefined"
      data-toggle="modal"
      type="button"
    >
      <i
        class="fas fa-pencil-alt"
      />
    </button>
  </div>
  <div
    aria-hidden="true"
    aria-labelledby="editRecipeModal"
    class="modal fade"
    id="edit-recipe-modal-undefined"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-lg"
      role="document"
    >
      <div
        class="modal-content"
      >
        <form
          id="edit-recipe-modal-form"
          name="editrecipeForm"
        >
          <div
            class="modal-header"
          >
            <input
              class="form-control recipetitle"
              name="recipetitle"
              onChange={[Function]}
              placeholder="Recipe Title"
              type="text"
            />
            <button
              aria-label="Close"
              class="close"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
          <div
            class="modal-body"
          >
            <h4>
              Ingredients
            </h4>
            <ul
              id="ingredients"
            />
            <button
              className="col-auto btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              <i
                class="fas fa-plus"
              />
            </button>
            <button
              className="col-auto btn btn-warning"
              onClick={[Function]}
              type="button"
            >
              clear
            </button>
            <h4>
              Directions
            </h4>
            <div
              class="row"
            >
              <div
                class="col-12"
              >
                <textarea
                  class="form-control directions"
                  name="directions"
                  onChange={[Function]}
                  rows="3"
                />
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-secondary"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
            <button
              class="btn btn-primary"
              data-dismiss="modal"
              onClick={[Function]}
              type="submit"
            >
              Save changes
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
